medicines/web/src/components/mip/index.tsx

```diff
@@ -119,17 +121,41 @@ const Mip: React.FC = () => {
     });
   };
 
+  const handleDocTypeCheckbox = async (d: DocType) => {
+    const a = Array.from(docTypes);
+    if (a.includes(d)) {
+      const docTypeIndex = a.indexOf(d);
+      a.splice(docTypeIndex, 1);
+    } else {
+      a.push(d);
+    }
+    setDocTypes(a);
+  };
+
   const rerouteSearchResults = (pageNo: number) => {
     router.push({
       pathname: router.route,
-      query: { search, page: pageNo },
+      query: {
+        search,
+        page: pageNo,
+        doc: docTypes.length > 0 ? docTypes.join(',') : null,
+      },
     });
   };
 
+  useEffect(() => {
+    rerouteSearchResults(1);
+  }, [docTypes]);
+
   useEffect(() => {
     if (searchTerm && page) {
       if (typeof searchTerm === 'string') {
         (async () => {
+          let d = null;
```

Comment:
```js
let docType = (typeof queryDocFilter === 'string') 
  ? formatDocTypeFilters(queryDocFilters) 
  : null;
```

Reply:
thanks, have refactored this a bit to use the ternary and use the result for a check on whether to call the `useState` setter afterwards